### DISCUSSION TITLE
Changes to Contract (MVP)

### DISCUSSION THREAD

**Discussion by thelostone-mc** [gitcoinco/grants-stack#419] [Q&A]
**RoundContract**

- [x] Add time-bound checks to vote function. 
This to ensure vote function can be invoked only during round running period
- [x] Discuss @nfrgosselin to add time bound checks to applyToRound
This to ensure applyToRound function can be invoked only during round application period
- [x] While creating a round -> round contract invokes setup function on the voting contract
This to set to store the round contract reference in the vote function (this will allow us to add a condition stating only the roun  can invoke the vote function and no-one else. This is currently a vulnerability in our design)

**VotingInterface**

- [x] Update VotingInterface to define a setup function called once to link voting to round contract (This refers to the last point. Adding this to the interface makes it clear that every strategy has to implement this and store this round contract in the voting contract

**QFStrategy**

- [x] Implement setup function in QFStrategy to link back to Round (refer to the setup function which will set the round address in the voting contract) 
- [ ] Update vote function to also handle native token transfer (should we support only ERC20 / is there an expectation to also support native chain token )
- [x] Ensure vote function can be invoked only by round contract (once the setup function is written up, we can add safeguard conditions to ensure only the round contract can invoke the vote contract)

**Comment by vacekj** [comment]
Closing as stale.
